packages/babel-plugin-transform-typescript/src/namespace.ts

```diff
@@ -185,6 +185,13 @@ function handleNested(
       case "TSEnumDeclaration":
       case "FunctionDeclaration":
       case "ClassDeclaration": {
+        if (
+          subNode.declaration.type === "ClassDeclaration" &&
+          subNode.declaration.declare
+        ) {
+          continue;
+        }
+
```

Comment:
`subNode.declaration.type === "ClassDeclaration"` seems to be a duplicate.

Reply:
This is not necessary, you are right.